Samples/excel-add-in-chart-analyze-data/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,588 @@
+      chart.top = 20;
+      chart.width = 300;
+      chart.setPosition("Q16", "V30");
+      chart.legend.visible = false;
+
+      let axis = chart.axes.categoryAxis;
+      let yaxis = chart.axes.valueAxis;
+      yaxis.displayUnit = "Thousands";
+      let categoryNameRange = sheet.getRange("A2:A33");
+      axis.setCategoryNames(categoryNameRange);
+
+      let trendlines = chart.series.getItemAt(0).trendlines;
+      trendlines.add("MovingAverage");
+      let tre = trendlines.getItem(0);
+      tre.movingAveragePeriod = 7;
+      showStatus('Success for "Overall sales trend"', false);
+    });
+  } catch (error) {
+    showStatus(error, true);
+  }
+}
+
+async function clearDatalables() {
+  try {
+    await Excel.run(async (context) => {
+      const sheet = context.workbook.worksheets.getItem("Sample");
+      let chart = sheet.charts.getItem("overallSales");
+      let points = chart.series.getItemAt(0).points;
+      points.load();
+      await context.sync();
+      let count = points.count;
+      for (let i = 0; i < count; i++) {
+        let point = points.getItemAt(i);
+        point.hasDataLabel = false;
+      }
+      await context.sync();
+    });
+  } catch (error) {
+    showStatus(error, true);
+  }
+}
+
+//add a Axis Title to the Category Axis of the the first chart in the worksheet
+async function highlightSales() {
+  try {
+    await Excel.run(async (context) => {
+      showStatus("Running", false);
+      const sheet = context.workbook.worksheets.getItem("Sample");
+      let chart = sheet.charts.getItem("overallSales");
+      let points = chart.series.getItemAt(0).points;
+      points.load();
+      await context.sync();
+      let count = points.count;
+      let max = 0;
+      //let index = 0;
+      for (let i = 0; i < count; i++) {
+        let point = points.getItemAt(i);
+        //point.hasDataLabel = false;
+        point.load("value");
+        await context.sync();
+        if (point.value > max) {
+          index = i;
+          max = point.value;
+        }
+      }
+      clearDatalables();
+      highlight();
+      await context.sync();
+      showStatus('Success for "Highlight highest sales"', false);
+    });
+  } catch (error) {
+    showStatus(error, true);
+  }
+}
+
+async function highlight() {
+  try {
+    await Excel.run(async (context) => {
+      const sheet = context.workbook.worksheets.getItem("Sample");
+      let chart = sheet.charts.getItem("overallSales");
+      chart.load("series");
+      await context.sync();
+      let points = chart.series.getItemAt(0).points;
+      let maxPoint = points.getItemAt(index);
+      maxPoint.hasDataLabel = true;
+      let datalabel = maxPoint.dataLabel;
+      datalabel.showCategoryName = true;
+      datalabel.showValue = true;
+      datalabel.showLegendKey = true;
+
+      maxPoint.markerStyle = "Diamond";
+      await context.sync();
+    });
+  } catch (error) {
+    showStatus(error, true);
+  }
+}
+
+//add Sample Data
```

Comment:
If the comment doesn't add more information than the function name, you can get rid of it.
```suggestion
```